sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,239 @@
+
+        c_joint_point = c_masscenter.locatenew(
+            c_name + '_child_joint',
+            c_frame.x + c_frame.y + c_frame.z)
+
+        c_joint_point.set_pos(p_joint_point, 0)
+        c_masscenter.v2pt_theory(p_masscenter, p_frame, c_frame)
+
+        # JointsMethod
+        q_ind = [theta]
+        u_ind = [omega]
+        kd = [thetad - omega]
+        BL = [parent, child]
+        gravity = Symbol('gravity')
+        FL = [(c_masscenter, c_mass * gravity * c_frame.y)]
+
+        KM = KanesMethod(p_frame, q_ind=q_ind, u_ind=u_ind, kd_eqs=kd)
+        (fr, frstar) = KM.kanes_equations(FL, BL)
+        eom = KM.rhs()
+
+        assert eom == self.eom
+
+
+class TestSlidingJoint():
+    def setup(self):
+        parent = Body('parent')
+        child = Body('child')
+
+        gravity = Symbol('gravity')
+        child.add_force((child.get_mass() * gravity, 0, 0), child.get_masscenter())
+
+        sliding_joint = SlidingJoint('sliding_joint', parent, child,
+                                     parent_point_pos=(1, 1, 1),
+                                     child_point_pos=(1, 1, 1))
+
+        spring_constant = Symbol('k')
+        child.add_force((spring_constant * sliding_joint.get_coordinates()[0], 0, 0),
+                        child.get_masscenter())
+
+        JM = JointsMethod([sliding_joint], parent)
+        self.eom = JM.rhs()
+
+    def test_simple_equations_of_motion(self):
+        # parent
+        p_name = "parent"
+        p_frame = ReferenceFrame(p_name + '_frame')
+        p_masscenter = Point(p_name + '_masscenter')
+        p_masscenter.set_vel(p_frame, 0)
+        p_mass = Symbol(p_name + '_mass')
+        p_inertia = (inertia(p_frame, 1, 1, 1), p_masscenter)
+        parent = RigidBody(p_name, p_masscenter, p_frame, p_mass, p_inertia)
+
+        # child
+        c_name = "child"
+        c_frame = ReferenceFrame(c_name + '_frame')
+        c_masscenter = Point(c_name + '_masscenter')
+        c_masscenter.set_vel(c_frame, 0)
+        c_mass = Symbol(c_name + '_mass')
+        c_inertia = (inertia(c_frame, 1, 1, 1), c_masscenter)
+        child = RigidBody(c_name, c_masscenter, c_frame, c_mass, c_inertia)
+
+        # sliding joint
+        dis = dynamicsymbols('dis')
+        disd = dynamicsymbols('dis', 1)
+        vel = dynamicsymbols('vel')
+
+        p_direction = p_frame.x
+        c_direction = c_frame.x
+
+        c_frame.orient(p_frame, 'Axis', [0, p_frame.x])
+
+        p_joint_point = p_masscenter.locatenew(
+            p_name + '_parent_joint',
+            p_frame.x + p_frame.y + p_frame.z)
+
+        c_joint_point = c_masscenter.locatenew(
+            c_name + '_child_joint',
+            c_frame.x + c_frame.y + c_frame.z)
+
+        p_joint_point.set_vel(p_frame, 0)
+        c_joint_point.set_vel(c_frame, 0)
+
+        c_joint_point.set_pos(p_joint_point, dis * p_direction)
+        c_joint_point.set_vel(p_frame, vel * p_direction)
+        c_masscenter.set_vel(p_frame, vel * p_direction)
```

Comment:
You should be setting the positions and velocities of all of the mass points. What about p_masscenter?